cns/restserver/internalapi_test.go

```diff
@@ -359,6 +397,29 @@ func validateNCStateAfterReconcile(t *testing.T, ncRequest *cns.CreateNetworkCon
 	}
 }
 
+func createNCReqInternal(t *testing.T, secondaryIPConfigs map[string]cns.SecondaryIPConfig, ncID, ncVersion string) cns.CreateNetworkContainerRequest {
+	req := generateNetworkContainerRequest(secondaryIPConfigs, ncID, ncVersion)
+	returnCode := svc.CreateOrUpdateNetworkContainerInternal(req, fakes.NewFakeScalar(releasePercent, requestPercent, batchSize), fakes.NewFakeNodeNetworkConfigSpec(initPoolSize))
+	if returnCode != 0 {
+		t.Fatalf("Failed to createNetworkContainerRequest, req: %+v, err: %d", req, returnCode)
+	}
+	return req
+}
+
+func validateSecondaryIPsNCVersion(t *testing.T, req cns.CreateNetworkContainerRequest) {
+	containerStatus := svc.state.ContainerStatus[req.NetworkContainerid]
+	// Validate secondary IPs' NC version has been updated by NC request
+	ncVersion, _ := strconv.Atoi(containerStatus.VMVersion)
+	for _, secIPConfig := range containerStatus.CreateNetworkContainerRequest.SecondaryIPConfigs {
+		// Though "10.0.0.16" IP exists in NC version 1, secodanry IP still keep its original NC version 0
+		if (secIPConfig.IPAddress == "10.0.0.16" && secIPConfig.NCVersion != 0) ||
+			(secIPConfig.IPAddress == "10.0.0.17" && secIPConfig.NCVersion != 1) {
+			t.Fatalf("nc request version is %d, secondary ip %s nc version is %d, expected nc version is 0",
+				ncVersion, secIPConfig.IPAddress, secIPConfig.NCVersion)
```

Comment:
These assertion values (the expected ip addresses and versions) are very far from the test. If you need to create similar tests with different IP addresses, this would not be very reusable. You should define the expected values on the test, and pass them to this function. The check is also very narrow: if there was only a secondary IP of 10.0.0.15 for example, this would pass.

Reply:
No. This test is designed specifically and is validating specific scenario.
In this specific test, secondary IP address only contains 10.0.0.16 and 10.0.0.17.
10.0.0.16 must be version 0 and 10.0.0.17 must be version 1.